File: src/main/kotlin/com/miro/miroappoauth/dto/LineType.java
package com.miro.miroappoauth.dto;

import org.jetbrains.annotations.Nullable;

import java.util.stream.Stream;

public enum LineType {
    straight,
    orthogonal,
    bezier,
    sketch;

    public static LineType fromString(@Nullable String value) {
        return Stream.of(values())
                .filter(cnst -> cnst.name().equalsIgnoreCase(value))
                .findFirst()
                .orElse(straight);
    }
}

File: src/main/kotlin/com/miro/miroappoauth/client/v1/LineType.kt
package com.miro.miroappoauth.client.v1

import java.util.stream.Stream

enum class LineType {
    straight,
    orthogonal,
    bezier,
    sketch;

    companion object {
        fun fromString(value: String?): LineType {
            return Stream.of(*values())
                .filter { cnst: LineType -> cnst.name.equals(value, ignoreCase = true) }
                .findFirst()
                .orElse(straight)
        }
    }
}
